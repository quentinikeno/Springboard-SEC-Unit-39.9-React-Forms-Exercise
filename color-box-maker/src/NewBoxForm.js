import React from "react";

const NewBoxForm = ({ addBox }) => {
	const INITIAL_STATE = { width: "", height: "", backgroundColor: "" };
	const [formData, setFormData] = INITIAL_STATE;
	const handleChange = (event) => {
		const { name, value } = event.target;
		setFormData((formData) => ({ ...formData, [name]: value }));
	};
	const handleSubmit = (event) => {
		event.preventDefault();
		addBox({ ...formData });
		setFormData(INITIAL_STATE);
	};
	return (
		<form onSubmit={handleSubmit}>
			<label htmlFor="width">Width</label>
			<input
				id="width"
				type="range"
				name="width"
				min="1"
				max="500"
				value={formData.width}
				onChange={handleChange}
			/>
			<span>{formData.width} pixels</span>
			<label htmlFor="height">Width</label>
			<input
				id="height"
				type="range"
				name="height"
				min="1"
				max="500"
				value={formData.height}
				onChange={handleChange}
			/>
			<span>{formData.height} pixels</span>
			<label htmlFor="backgroundColor">Width</label>
			<input
				id="backgroundColor"
				type="text"
				name="backgroundColor"
				placeholder="Background Color"
				value={formData.backgroundColor}
				onChange={handleChange}
			/>
		</form>
	);
};

export default NewBoxForm;
