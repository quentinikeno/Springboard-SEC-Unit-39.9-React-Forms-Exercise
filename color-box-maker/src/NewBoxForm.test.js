import { render } from "@testing-library/react";
import NewBoxForm from "./NewBoxForm";

test("rendering without crashing", () => {
	render(<NewBoxForm />);
});

test("matching a snapshot", () => {
	const { asFragment } = render(<NewBoxForm />);
	expect(asFragment()).toMatchSnapshot();
});
