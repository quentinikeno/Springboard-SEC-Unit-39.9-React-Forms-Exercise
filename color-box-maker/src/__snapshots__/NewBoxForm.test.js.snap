// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot 1`] = `
<DocumentFragment>
  <form>
    <div>
      <label
        for="width"
      >
        Width
      </label>
      <input
        id="width"
        max="500"
        min="1"
        name="width"
        type="range"
        value="250"
      />
      <span>
        250 pixels
      </span>
    </div>
    <div>
      <label
        for="height"
      >
        Height
      </label>
      <input
        id="height"
        max="500"
        min="1"
        name="height"
        type="range"
        value="250"
      />
      <span>
        250 pixels
      </span>
    </div>
    <div>
      <label
        for="backgroundColor"
      >
        Background Color
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        placeholder="Background Color"
        type="text"
        value=""
      />
    </div>
    <button>
      Create a new box!
    </button>
  </form>
</DocumentFragment>
`;
